db/upgrade.php

```diff
@@ -13,57 +13,109 @@
-
-        // Conditionally launch create table for tool_mergeusersr
+        // Conditionally launch create table for tool_mergeusers.
         if (!$dbman->table_exists($table)) {
             $dbman->create_table($table);
-        }
-
-        // mergeusers savepoint reached
+        }        
+        // Mergeusers savepoint reached.
         upgrade_plugin_savepoint(true, 2013112912, 'tool', 'mergeusers');
     }
-
+    if ($oldversion < 2023040460) {
+        // Define table tool_mergeusers_queue to be created.
+        $table = new xmldb_table('tool_mergeusers_queue');
+        // Adding fields to table tool_mergeusers_queue.
+        $table->add_field('id', XMLDB_TYPE_INTEGER, '10', null, XMLDB_NOTNULL, XMLDB_SEQUENCE, null);
+        $table->add_field('removeuserfield', XMLDB_TYPE_CHAR, '50', null, XMLDB_NOTNULL, null, null);
+        $table->add_field('removeuservalue', XMLDB_TYPE_CHAR, '255', null, XMLDB_NOTNULL, null, null);
+        $table->add_field('removeuserid', XMLDB_TYPE_INTEGER, '10', null, null, null, null);
+        $table->add_field('keepuserfield', XMLDB_TYPE_CHAR, '50', null, XMLDB_NOTNULL, null, null);
+        $table->add_field('keepuservalue', XMLDB_TYPE_CHAR, '255', null, XMLDB_NOTNULL, null, null);
+        $table->add_field('keepuserid', XMLDB_TYPE_INTEGER, '10', null, null, null, null);
+        $table->add_field('timeadded', XMLDB_TYPE_INTEGER, '10', null, XMLDB_NOTNULL, null, null);
+        $table->add_field('timemodified', XMLDB_TYPE_INTEGER, '10', null, null, null, null);
+        $table->add_field('status', XMLDB_TYPE_INTEGER, '4', null, XMLDB_NOTNULL, null, null);
+        $table->add_field('retries', XMLDB_TYPE_INTEGER, '4', null, XMLDB_NOTNULL, null, null);
+        $table->add_field('log', XMLDB_TYPE_TEXT, null, null, null, null, null);
+        // Adding keys to table tool_mergeusers_queue.
+        $table->add_key('primary', XMLDB_KEY_PRIMARY, ['id']);
+        // Adding indexes to table tool_mergeusers_queue.
+        $table->add_index('mdl_toolmerg_ruv_ix', XMLDB_INDEX_NOTUNIQUE, ['removeuservalue']);
+        $table->add_index('mdl_toolmerg_kuv_ix', XMLDB_INDEX_NOTUNIQUE, ['keepuservalue']);
+        $table->add_index('mdl_toolmerg_tcs_ix', XMLDB_INDEX_NOTUNIQUE, ['timeadded', 'status']);
+        $table->add_index('mdl_toolmerg_tms_ix', XMLDB_INDEX_NOTUNIQUE, ['timemodified', 'status']);
+        // Conditionally launch create table for tool_mergeusers_queue and populate it importing data from old table.
+        if (!$dbman->table_exists($table)) {
+            $dbman->create_table($table);
+            // Export data from mergeusers table to the new one tool_mergeusers_queue.
+            $filter=null;
+            $limitfrom=0;
+            $limitnum=0; 
+            $sort = " id ASC";
+            $records = $DB->get_records(merge_request::TABLE_MERGE_REQUEST_OLD, $filter, $sort, 
```

Comment:
I think this code is the same like in the class merge_request. We should be using that from the merge_request class, I think. I should be save too.

Reply:
done